dm/pkg/dumpling/utils.go

```diff
@@ -200,3 +205,114 @@ func ParseFileSize(fileSizeStr string, defaultSize uint64) (uint64, error) {
+	dumplingFlagSet.Uint64VarP(&dumpCfg.StatementSize, "statement-size", "s", dumpCfg.StatementSize, "Attempted size of INSERT statement in bytes")
+	dumplingFlagSet.StringVar(&dumpCfg.Consistency, "consistency", dumpCfg.Consistency, "Consistency level during dumping: {auto|none|flush|lock|snapshot}")
+	dumplingFlagSet.StringVar(&dumpCfg.Snapshot, "snapshot", dumpCfg.Snapshot, "Snapshot position. Valid only when consistency=snapshot")
+	dumplingFlagSet.BoolVarP(&dumpCfg.NoViews, "no-views", "W", dumpCfg.NoViews, "Do not dump views")
+	dumplingFlagSet.Uint64VarP(&dumpCfg.Rows, "rows", "r", dumpCfg.Rows, "Split table into chunks of this many rows, default unlimited")
+	dumplingFlagSet.StringVar(&dumpCfg.Where, "where", dumpCfg.Where, "Dump only selected records")
+	dumplingFlagSet.BoolVar(&dumpCfg.EscapeBackslash, "escape-backslash", dumpCfg.EscapeBackslash, "Use backslash to escape quotation marks")
+	dumplingFlagSet.StringArrayVarP(&filters, "filter", "f", []string{"*.*"}, "Filter to select which tables to dump")
+	dumplingFlagSet.StringVar(&dumpCfg.Security.CAPath, "ca", dumpCfg.Security.CAPath, "The path name to the certificate authority file for TLS connection")
+	dumplingFlagSet.StringVar(&dumpCfg.Security.CertPath, "cert", dumpCfg.Security.CertPath, "The path name to the client certificate file for TLS connection")
+	dumplingFlagSet.StringVar(&dumpCfg.Security.KeyPath, "key", dumpCfg.Security.KeyPath, "The path name to the client private key file for TLS connection")
+	dumplingFlagSet.BoolVar(&noLocks, "no-locks", false, "")
+	dumplingFlagSet.BoolVar(&dumpCfg.TransactionalConsistency, "transactional-consistency", true, "Only support transactional consistency")
+
+	err := dumplingFlagSet.Parse(args)
+	if err != nil {
+		return err
+	}
+
+	// compatibility for `--no-locks`
+	if noLocks {
+		logger.Warn("`--no-locks` is replaced by `--consistency none` since v2.0.0")
+		// it's default consistency or by meaning of "auto", we could overwrite it by `none`
+		if dumpCfg.Consistency == "auto" {
+			dumpCfg.Consistency = "none"
+		} else if dumpCfg.Consistency != "none" {
+			return errors.New("cannot both specify `--no-locks` and `--consistency` other than `none`")
+		}
+	}
+
+	if fileSizeStr != "" {
+		dumpCfg.FileSize, err = ParseFileSize(fileSizeStr, export.UnspecifiedSize)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(tablesList) > 0 || (len(filters) != 1 || filters[0] != "*.*") {
+		ff, err2 := parseTableFilter(tablesList, filters)
+		if err2 != nil {
+			return err2
+		}
+		dumpCfg.TableFilter = ff // overwrite `block-allow-list`.
+		logger.Warn("overwrite `block-allow-list` by `tables-list` or `filter`")
+	}
+
+	return nil
+}
+
+// parseTableFilter parses `--tables-list` and `--filter`.
+// copy (and update) from https://github.com/pingcap/dumpling/blob/6f74c686e54183db7b869775af1c32df46462a6a/cmd/dumpling/main.go#L222.
+func parseTableFilter(tablesList, filters []string) (tfilter.Filter, error) {
```

Comment:
seems we can `ParseTableFilter` in  `tidb/dumpling/export/config.go`,  so i perfer to delete this func

Reply:
lgtm